src/data_gradients/feature_extractors/classification/class_distribution.py

```diff
@@ -0,0 +1,79 @@
+import collections
+
+import pandas as pd
+
+from data_gradients.common.registry.registry import register_feature_extractor
+from data_gradients.feature_extractors.abstract_feature_extractor import Feature
+from data_gradients.utils.data_classes.data_samples import ClassificationSample
+from data_gradients.visualize.seaborn_renderer import BarPlotOptions
+from data_gradients.feature_extractors.abstract_feature_extractor import AbstractFeatureExtractor
+
+
+@register_feature_extractor()
+class ClassificationClassDistribution(AbstractFeatureExtractor):
+    """Feature Extractor to count the number of labels of each class."""
+
+    def __init__(self):
+        self.data = []
+
+    def update(self, sample: ClassificationSample):
+        class_name = sample.class_names[sample.class_id]
+        self.data.append(
+            {
+                "split": sample.split,
+                "class_id": sample.class_id,
+                "class_name": class_name,
+            }
+        )
+
+    def aggregate(self) -> Feature:
+        df = pd.DataFrame(self.data)
+
+        # Include ("class_name", "split", "n_appearance")
+        df_class_count = df.groupby(["class_name", "class_id", "split"]).size().reset_index(name="n_appearance")
+        all_class_names = df_class_count["class_name"].unique()
+
+        # Height of the plot is proportional to the number of classes
+        n_unique = len(all_class_names)
+        figsize_x = 10
+        figsize_y = min(max(6, int(n_unique * 0.3)), 175)
+
+        plot_options = BarPlotOptions(
+            x_label_key="n_appearance",
+            x_label_name="Class support (# of samples)",
```

Comment:
Not sure if I'm the only one, but I feel like `#` in `Class support (# of samples)` looks kinda dirty in the context of a report